platform/python/template_init.py

```diff
@@ -0,0 +1,10 @@
+# this is a template file that gets instantiated for any bazel package with python-accessible resources.
+# it provides a convenience function, but its simple presence as __init__.py is enough to denote a package's existence.
```

Comment:
I'm not sure I understand the resulting python package structure, can you explain it?

Reply:
py_binary produces a package structure that corresponds directly to the bazel source tree. See https://github.com/sipb/homeworld/pull/487#discussion_r379573921.

The way the PYTHONPATH is set up lets us access particular things in a few ways:

 - `import dnsmasq` imports the `//dnsmasq` python package (via its `__init__.py` file, which is therefore mandatory for the package to be usable by python)
 - similarly, `import spire.resources` refers to the `//spire/resources` python package.
 - we can also import python modules in the `//spire:src` directory, because that's where the source code comes from.
 - we would be able to import `__main__.spire.resources` as well, except that `__main__` is interpreted as a special package in python and cannot be used like that. but we could use this if we wanted to reference something built in another workspace, like `@ceph//...` or something.

So the idea of template_init.py, as instantiated by py_resources into many copies of `__init__.py`, is to make it possible for us to access resources from a variety of bazel packages by turning them into stub python packages. This means that `pkgutil.get_data` can work, because they can be imported. We could manually do something around figuring out where the runfiles directory is stored, and extracting files directly from it, but this seems somewhat cleaner?